java-checks/src/main/java/org/sonar/java/checks/IfElseIfStatementEndsWithElseCheck.java

```diff
@@ -37,11 +37,14 @@
 
   @Override
   public void visitNode(Tree tree) {
-    StatementTree elseStmt = ((IfStatementTree) tree).elseStatement();
-    if (elseStmt != null
-      && elseStmt.is(Tree.Kind.IF_STATEMENT) && ((IfStatementTree) elseStmt).elseStatement() == null) {
-
-      reportIssue(((IfStatementTree) elseStmt.parent()).elseKeyword(), ((IfStatementTree) elseStmt).ifKeyword(), "\"if ... else if\" constructs should end with \"else\" clauses.");
+    IfStatementTree treeIfStmt = (IfStatementTree) tree;
+    StatementTree elseStmt = treeIfStmt.elseStatement();
+    IfStatementTree ifStmt = null;
+    if (elseStmt != null && elseStmt.is(Tree.Kind.IF_STATEMENT)) {
+      ifStmt = (IfStatementTree) elseStmt;
```

Comment:
I think you are over-complicating things here. Try to keep it simple and "logical" to read, introducing the variable you need and minimizing the number of validation you need to do.

In this case, simply nest the 2 `if` statements, and put the definition of the `ifStmt` variable within the body of the first `if`. you can then remove the `ifStmt != null` test.